extra/mariabackup/backup_copy.cc

```diff
@@ -2068,6 +2072,47 @@ bool backup_files_from_datadir(ds_ctxt_t *ds_data,
 	return ret;
 }
 
+bool backup_mroonga_files_from_datadir(ds_ctxt_t *ds_data,
+                                       const char *dir_path)
+{
+	os_file_dir_t dir= os_file_opendir(dir_path);
+	if (dir == IF_WIN(INVALID_HANDLE_VALUE, nullptr)) return false;
+
+	os_file_stat_t info;
+	bool ret= true;
+	while (os_file_readdir_next_file(dir_path, dir, &info) == 0)
+	{
+
+		if (info.type != OS_FILE_TYPE_FILE)
+			continue;
+
+		const char *pname = strrchr(info.name, '/');
+#ifdef _WIN32
+		if (const char *last = strrchr(info.name, '\\'))
+		{
+			if (!pname || last > pname)
+				pname = last;
+		}
+#endif
+		if (!pname)
+			pname = info.name;
+
+		if (!strstr(pname, ".mrn"))
```

Comment:
Can't you just use  `ends_with(info.name, ".mrn")` ,  instead of manipulations with pname? We only want to check file extension, right? and `ends_with` is  used exactly for it.

Reply:
Unfortunately, mroonga files have filenames like
```
table.mrn
table.mrn.000000
table.mrn.000100
table.mrn.001
table.mrn.conf
table.mrn.options
```

/cc @kou @komainu8

Reply:
@fdiary is (almost) correct. `database.mrn*` not `table.mrn*`. If we know the target database name here, we may be able to use `starts_with(info.name, "${database_name}.mrn")`.